mypyc/irbuild/ll_builder.py

```diff
@@ -592,7 +601,12 @@ def make_dict(self, key_value_pairs: Sequence[DictEntry], line: int) -> Value:
                 )
 
         if result is None:
-            result = self.primitive_op(new_dict_op, initial_items, line)
+            items_size = len(initial_items)
+            if items_size > 0:
+                size = self.add(LoadInt(items_size // 2, -1, c_int_rprimitive))
+                result = self.call_c(dict_build_op, [size] + initial_items, line)
+            else:
+                result = self.call_c(dict_new_op, [], line)
```

Comment:
This same code is repeated above around line 589. Maybe extract these lines into a helper method and share the code?